cookbooks/aws-parallelcluster-config/files/default/custom_action_executor/custom_action_executor.py

```diff
@@ -62,12 +62,115 @@ class ExecutableScript(ScriptDefinition):
     path: str
 
 
+class LegacyEventName(Enum):
+    """Maps legacy events names to avoid changing script contract."""
+
+    ON_NODE_START = "preinstall"
+    ON_NODE_CONFIGURED = "postinstall"
+    ON_NODE_UPDATED = "postupdate"
+
+    def map_to_current_name(self):
+        """Return the current event name value as it's configured in the cluster config."""
+        if self == LegacyEventName.ON_NODE_START:
+            result = "OnNodeStart"
+        elif self == LegacyEventName.ON_NODE_CONFIGURED:
+            result = "OnNodeConfigured"
+        elif self == LegacyEventName.ON_NODE_UPDATED:
+            result = "OnNodeUpdated"
+        else:
+            raise ValueError(f"Unknown legacy event name: {self.value}")
+
+        return result
+
+    def __str__(self):
+        """Return the legacy event name value."""
+        return self.value
+
+
+@dataclass
+class CustomActionsConfig:
+    """
+    Encapsulates custom actions configuration.
+
+    Contains all the configuration relevant to custom actions execution.
+    """
+
+    stack_name: str
+    region_name: str
+    node_type: str
+    queue_name: str
+    resource_name: str
+    instance_id: str
+    instance_type: str
+    ip_address: str
+    hostname: str
+    availability_zone: str
+    scheduler: str
+    event_name: str
+    legacy_event: LegacyEventName
+    can_execute: bool
+    dry_run: bool
+    script_sequence: list  # type list[ScriptDefinition]
+    script_sequences_per_event: dict
+    event_file_override: str
+    node_spec_file: str
+
+
+class EnvEnricher:
+    """Provide the environment variables for the ExecutableScript."""
+
+    def __init__(self):
+        self._base_env = os.environ.copy()
+
+    def build_env(self, exe_script: ExecutableScript):  # pylint: disable=unused-argument
+        """Provide a copy of the current process environment variables."""
+        return self._base_env.copy()
+
+
+class CfnConfigEnvEnricher(EnvEnricher):
+    """Provides cnfconfig backward compatible environment variables for the ExecutableScript."""
+
+    def __init__(self, conf: CustomActionsConfig):
+        super().__init__()
+        self.conf = conf
+        self._cfn_base_env = self._create_additional_cfnconfig_compatible_env()
```

Comment:
I'm not sure, but I think I've seen a similar pattern get flagged by the linters (but I could be wrong). Would it make sense to make both of the below calls static and pass self.conf as a parameter. Just to make sure there are no order of operation issues?

Reply:
You are right and I don't like it either. Let's see what I can do about it.